File: WebApplication/src/app-clickhouse/dtos/ch-query.dto.ts
import { TemplateSubquerySearchDto } from "../../app-templates/dtos/search/template-subquery-search.dto";
import { QueryType } from "../enums/query-type.enum";
import { ChPaginationDto } from "./ch-pagination.dto";
import { ChTableDto } from "./ch-table.dto";
import { ChConditionDto } from "./condition/ch-condition.dto";
import { ChGroupByDto } from "./group-by/ch-group-by.dto";
import { ChOrderByDto } from "./order-by/ch-order-by.dto";
import { ChOutputDto } from "./output/ch-output.dto";

export class ChQueryDto {
    queryType = QueryType.fromTable;
    table: ChTableDto;
    subquery: TemplateSubquerySearchDto;
    pagination = new ChPaginationDto();
    output: ChOutputDto[] = [];
    condition: ChConditionDto[] = [];
    groupBy: ChGroupByDto[] = [];
    orderBy: ChOrderByDto[] = [];
}

File: Dw.ClickHouse/Dtos/ChQueryDto.cs
using Ch.Models.Dtos.Condition;
using Ch.Models.Dtos.GroupBy;
using Ch.Models.Dtos.OrderBy;
using Ch.Models.Dtos.Output;
using Ch.Models.Enums;

namespace Ch.Models.Dtos
{
    public class ChQueryDto
    {
        public QueryType QueryType { get; set; } = QueryType.FromTable;
        public ChTableDto Table { get; set; }
        public TemplateSubquerySearchDto Subquery { get; set; }
        public ChPaginationDto Pagination { get; set; } = new ChPaginationDto();
        public List<ChOutputDto> Output { get; set; } = new List<ChOutputDto>();
        public List<ChConditionDto> Condition { get; set; } = new List<ChConditionDto>();
        public List<ChGroupByDto> GroupBy { get; set; } = new List<ChGroupByDto>();
        public List<ChOrderByDto> OrderBy { get; set; } = new List<ChOrderByDto>();
    }
}
